manifest.go

```diff
@@ -292,10 +307,15 @@ func (b *ManifestV2Builder) KeyMetadata(km []byte) *ManifestV2Builder {
 // a pointer to the constructed manifest file. Further calls to the modifier
 // methods after calling build would modify the constructed ManifestFile.
 func (b *ManifestV2Builder) Build() ManifestFile {
+	if b.m.partitionType == nil {
+		b.m.partitionType = UnpartitionedSpec.PartitionType(nil)
+	}
+
 	return b.m
 }
 
 type manifestFileV2 struct {
+	partitionType      *StructType
```

Comment:
Could this be integrated with the `FieldSummary` instead of being at the top level here?

Reply:
In either case, we should probably be populating this after reading in a manifest file too, right?

Reply:
Not sure if it belongs to `FieldSummary`. 
We also need to be able to do `Avro` -> `Iceberg` schema conversion to populate the field on read.

Reply:
I have done `Avro` <--> `Iceberg` schema conversions.
Can you please help me with the populate part ? Maybe point me to the right part of the code ?